### PR TITLE
Added runbook, basic alerts and rules

### LINKED ISSUE #240: Create Alerts and Runbooks in Jsonnet
Alerts to add:
- SLO budget burn read/write alerts
- Unhealthy ingesters/compactors/distributors in the ring for too long
- Blocklist growing for too long

### PR DESCRIPTION
**Pull request by joe-elliott**
**What this PR does**:
Adds basic alerts, rules and runbook entries for Tempo

**Which issue(s) this PR fixes**:
Fixes #240 

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/tempo-mixin/runbook.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+This document should help with remediating operational issues in Tempo.
+
+## TempoRequestErrors
+## TempoRequestLatency
+
+Aside from obvious errors in the logs the only real lever you can pull here is scaling.  Use the Reads or Writes dashboard 
+to identify the component that is struggling and scale it up.  It should be noted that right now quickly scaling the 
+Ingester component can cause 404s on traces until they are flushed to the backend.  For safety you may only want to 
+scale one per hour.  However, if Ingesters are falling over, it's better to scale fast, ingest successfully and throw 404s 
+on query then to have an unstable ingest path.  Make the call!
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:12]: `then->than`

File: operations/tempo-mixin/runbook.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## TempoCompactorUnhealthy
+
+Tempo by default uses [Memberlist](https://github.com/hashicorp/memberlist) to persist the ring state between components.
+Occassionally this results in old components staying in the ring which particularly impacts compactors because they start
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:17]: `Occassionally->Occasionally `

File: operations/tempo-mixin/runbook.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+falling behind on the blocklist.  If this occurs port-foward to 3100 on a compactor and bring up `/compactor/ring`.  Use the
+"Forget" button to drop any unhealthy compactors.
+
+Note that this more of an art then a science: https://github.com/grafana/tempo/issues/142
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:21]: `then->than`

File: operations/tempo-mixin/runbook.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+## TempoDistributorUnhealthy
+
+Tempo by default uses [Memberlist](https://github.com/hashicorp/memberlist) to persist the ring state between components.
+Occassionally this results in old components staying in the ring which impacts distributors only if you are using the 
+global rate limiting strategy.  If this occurs port-foward to 3100 on a distributor and bring up `/distributor/ring`.  Use the
+"Forget" button to drop any unhealthy distributors.
+
+Note that this more of an art then a science: https://github.com/grafana/tempo/issues/142
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:30]: `then->than`

**Review by dgzlopes — COMMENTED** [review]
Nits. Typos/Grammar.

File: operations/tempo-mixin/runbook.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+falling behind on the blocklist.  If this occurs port-foward to 3100 on a compactor and bring up `/compactor/ring`.  Use the
+"Forget" button to drop any unhealthy compactors.
+
+Note that this more of an art then a science: https://github.com/grafana/tempo/issues/142
+
+## TempoDistributorUnhealthy
+
+Tempo by default uses [Memberlist](https://github.com/hashicorp/memberlist) to persist the ring state between components.
+Occassionally this results in old components staying in the ring which impacts distributors only if you are using the 
+global rate limiting strategy.  If this occurs port-foward to 3100 on a distributor and bring up `/distributor/ring`.  Use the
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:27]: `foward ->forward`

File: operations/tempo-mixin/runbook.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+to identify the component that is struggling and scale it up.  It should be noted that right now quickly scaling the 
+Ingester component can cause 404s on traces until they are flushed to the backend.  For safety you may only want to 
+scale one per hour.  However, if Ingesters are falling over, it's better to scale fast, ingest successfully and throw 404s 
+on query then to have an unstable ingest path.  Make the call!
+
+## TempoCompactorUnhealthy
+
+Tempo by default uses [Memberlist](https://github.com/hashicorp/memberlist) to persist the ring state between components.
+Occassionally this results in old components staying in the ring which particularly impacts compactors because they start
+falling behind on the blocklist.  If this occurs port-foward to 3100 on a compactor and bring up `/compactor/ring`.  Use the
```
Comment (dgzlopes) [inline operations/tempo-mixin/runbook.md:18]: `foward ->forward`

**Comment by joe-elliott** [comment]
@dgzlopes Thank you

Something tells me that as long as I'm working in this repo we'll need a spell checker of sorts.  Added #319.

**Review by annanay25 — COMMENTED** [review]
Nice work, alerts coming to Tempo soon 🎉 

I left a few comments on this PR, another addition that I felt would be valuable would be to add instructions on how alerts & rules can be generated from the mixin.. I have the commit here - https://github.com/annanay25/tempo/commit/6d482460d860733379607f7d27033730b40c59db

File: operations/tempo-mixin/alerts.libsonnet [hunk 7]
```diff
@@ -0,0 +1,74 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'tempo_alerts',
+        rules: [
+          {
+            alert: 'TempoRequestErrors',
+            expr: |||
+              100 * sum(rate(tempo_request_duration_seconds_count{status_code=~"5.."}[1m])) by (namespace, job, route)
```
[the remark this answers is not in the document]
Comment (joe-elliott) [reply operations/tempo-mixin/alerts.libsonnet:10]: By default all Cortex and Loki alerts have namespace and job in their grouping.

I do not know why.  Perhaps so it works in environments with no job label?

**Review by annanay25 — APPROVED** [review]

**Pull request merged**